src/ai/backend/manager/models/gql_models/image.py

```diff
@@ -733,18 +724,90 @@ async def mutate(
                 ],
             )
             if client_role != UserRole.SUPERADMIN:
-                customized_image_owner = (image_row.labels or {}).get(
-                    "ai.backend.customized-image.owner"
-                )
-                if (
-                    not customized_image_owner
-                    or customized_image_owner != f"user:{ctx.user['uuid']}"
-                ):
+                if not image_row.is_customized_by(ctx.user["uuid"]):
                     return ForgetImage(ok=False, msg="Forbidden")
-            await session.delete(image_row)
+            await image_row.mark_as_deleted(session)
             return ForgetImage(ok=True, msg="", image=ImageNode.from_row(image_row))
 
 
+class PurgeImage(graphene.Mutation):
+    """Added in 25.3.1."""
+
+    allowed_roles = (
+        UserRole.SUPERADMIN,
+        UserRole.ADMIN,
+        UserRole.USER,
+    )
+
+    class Arguments:
+        reference = graphene.String(required=True)
+        architecture = graphene.String(default_value=DEFAULT_IMAGE_ARCH)
+
+    image = graphene.Field(ImageNode)
+
+    @staticmethod
+    async def mutate(
+        root: Any,
+        info: graphene.ResolveInfo,
+        reference: str,
+        architecture: str,
+    ) -> PurgeImage:
+        log.info("purge image {0} by API request", reference)
+        ctx: GraphQueryContext = info.context
+        client_role = ctx.user["role"]
+
+        async with ctx.db.begin_session() as session:
+            image_row = await ImageRow.resolve(
+                session,
+                [
+                    ImageIdentifier(reference, architecture),
+                    ImageAlias(reference),
+                ],
+            )
+            if client_role != UserRole.SUPERADMIN:
+                if not image_row.is_customized_by(ctx.user["uuid"]):
+                    return PurgeImage(ok=False, msg="Forbidden")
+            await session.delete(image_row)
+            return PurgeImage(image=ImageNode.from_row(image_row))
+
+
+class PurgeImageById(graphene.Mutation):
+    """Added in 25.3.1."""
+
+    allowed_roles = (
+        UserRole.SUPERADMIN,
+        UserRole.ADMIN,
+        UserRole.USER,
+    )
+
+    class Arguments:
+        image_id = graphene.String(required=True)
+
+    image = graphene.Field(ImageNode)
+
+    @staticmethod
+    async def mutate(
+        root: Any,
+        info: graphene.ResolveInfo,
+        image_id: str,
+    ) -> PurgeImageById:
+        log.info("purge image {0} by API request", image_id)
+        _image_id = extract_object_uuid(info, image_id, "image")
```

Comment:
It would actually be less confusing if you changed the variable name to a different one.